File: server/models/Comptable.js
const mongoose = require('mongoose');
const bcrypt = require('bcryptjs');

const comptableSchema = new mongoose.Schema({
  id_comptable: {
    type: String,
    unique: true
  },
  nom: {
    type: String,
    required: true
  },
  prenom: {
    type: String,
    required: true
  },
  email: {
    type: String,
    unique: true,
    required: true
  },
  telephone: {
    type: String,
    required: true
  },
  mot_de_passe: {
    type: String,
    required: true
  },
  statut: {
    type: String,
    enum: ['actif', 'inactif'],
    default: 'actif'
  }
}, {
  timestamps: true
});

comptableSchema.pre('save', async function(next) {
  if (!this.isModified('mot_de_passe')) return next();
  this.mot_de_passe = await bcrypt.hash(this.mot_de_passe, 10);
  next();
});

comptableSchema.pre('validate', async function(next) {
  if (!this.id_comptable) {
    try {
      const dernier = await this.constructor.findOne(
        { id_comptable: { $regex: /^CPT\d+$/ } },
        {},
        { sort: { 'id_comptable': -1 } }
      );
      
      let numero = 1;
      if (dernier && dernier.id_comptable) {
        const match = dernier.id_comptable.match(/^CPT(\d+)$/);
        if (match) {
          numero = parseInt(match[1]) + 1;
        }
      }
      this.id_comptable = `CPT${numero.toString().padStart(3, '0')}`;
    } catch (error) {
      return next(error);
    }
  }
  next();
});

module.exports = mongoose.model('Comptable', comptableSchema); 

File: eduflex_mobile/lib/models/comptable.dart
class Comptable {
  final String id;
  final String idComptable;
  final String nom;
  final String prenom;
  final String email;
  final String telephone;
  final String statut;

  Comptable({
    required this.id,
    required this.idComptable,
    required this.nom,
    required this.prenom,
    required this.email,
    required this.telephone,
    required this.statut,
  });

  factory Comptable.fromJson(Map<String, dynamic> json) {
    return Comptable(
      id: json['_id'] ?? json['id'] ?? '',
      idComptable: json['id_comptable'] ?? '',
      nom: json['nom'] ?? '',
      prenom: json['prenom'] ?? '',
      email: json['email'] ?? '',
      telephone: json['telephone'] ?? '',
      statut: json['statut'] ?? 'actif',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'id_comptable': idComptable,
      'nom': nom,
      'prenom': prenom,
      'email': email,
      'telephone': telephone,
      'statut': statut,
    };
  }
}
